### PR TITLE
SNOW-683057: Improve session.py code coverage

### PR DESCRIPTION
**Pull request by sfc-gh-yixie**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-683057

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Use mock to add tests to the not-covered session.py

### CODE REVIEW

File: tests/unit/test_session.py [hunk 1]
```diff
@@ -79,3 +84,116 @@ def test_used_scoped_temp_object():
...
+    fake_connection._conn = mock.Mock()
+    session = Session(fake_connection)
+    with pytest.raises(
+        ValueError,
+        match="The provided schema or inferred schema cannot be None or empty",
+    ):
+        session.create_dataframe([[1]], schema=StructType([]))
+
+
+def test_create_dataframe_wring_type():
```
Comment (sfc-gh-aalam) [inline tests/unit/test_session.py:172]:
```suggestion
def test_create_dataframe_wrong_type():
```

File: tests/unit/test_session.py [hunk 2]
```diff
@@ -79,3 +84,116 @@ def test_used_scoped_temp_object():
...
+        if sql == "select system$are_anaconda_terms_acknowledged()":
+            return [[False]]
+
+    session.table = MagicMock(name="session.table")
+    session.table.side_effect = get_information_schema_packages
+    session._run_query = MagicMock(name="session._run_query")
+    session._run_query.side_effect = run_query
+    with pytest.raises(
+        ValueError,
+        match="Cannot add package random_package_name because Anaconda terms must be accepted by ORGADMIN to use Anaconda 3rd party packages. Please follow the instructions at https://docs.snowflake.com/en/developer-guide/udf/python/udf-python-packages.html#using-third-party-packages-from-anaconda.",
```
Comment (sfc-gh-aalam) [inline tests/unit/test_session.py:144]: this is a very long line. Should we break this down into a smaller one?

File: tests/unit/test_session.py [hunk 3]
```diff
@@ -79,3 +84,119 @@ def test_used_scoped_temp_object():
...
+    fake_connection._conn = mock.Mock()
+    session = Session(fake_connection)
+    with pytest.raises(
+        SnowparkInvalidObjectNameException,
+        match="The object name 'a.b.c.d' is invalid.",
+    ):
+        session._table_exists("a.b.c.d")
+
+
+def test_explain_query_error():
```
Comment (sfc-gh-jdu) [inline tests/unit/test_session.py:196]: this is covered in https://github.com/snowflakedb/snowpark-python/pull/594/files

**Review by sfc-gh-jdu — APPROVED** [review]

**Review by sfc-gh-sfan — APPROVED** [review]

**Review by sfc-gh-aalam — APPROVED** [review]

File: tests/unit/test_session.py (hunk 2 above)
Comment (sfc-gh-yixie) [reply tests/unit/test_session.py]: Broken into multi-lines

**Pull request merged**